docs/src/main/asciidoc/kafka-streams-guide.adoc

```diff
@@ -0,0 +1,1141 @@
+            Arrays.asList(
+                    new WeatherStation(1, "Hamburg", 13),
+                    new WeatherStation(2, "Snowdonia", 5),
+                    new WeatherStation(3, "Boston", 11),
+                    new WeatherStation(4, "Tokio", 16),
+                    new WeatherStation(5, "Cusco", 12),
+                    new WeatherStation(6, "Svalbard", -7),
+                    new WeatherStation(7, "Porthsmouth", 11),
+                    new WeatherStation(8, "Oslo", 7),
+                    new WeatherStation(9, "Marrakesh", 20)
+            ));
+
+
+    @Outgoing("temperature-values")                             // <1>
+    public Flowable<KafkaMessage<Integer, String>> generate() {
+
+        return Flowable.interval(500, TimeUnit.MILLISECONDS)    // <2>
+                .map(tick -> {
+                    WeatherStation station = stations.get(random.nextInt(stations.size()));
+                    double temperature = new BigDecimal(random.nextGaussian() * 15 + station.averageTemperature)
+                            .setScale(1, RoundingMode.HALF_UP)
+                            .doubleValue();
+
+                    LOG.info("station: {}, temperature: {}", station.name, temperature);
+                    return KafkaMessage.of(station.id, Instant.now() + ";" + temperature);
+                });
+    }
+
+    @Outgoing("weather-stations")                               // <3>
+    public Flowable<KafkaMessage<Integer, String>> weatherStations() {
+        List<KafkaMessage<Integer, String>> stationsAsJson = stations.stream()
+            .map(s -> KafkaMessage.of(s.id, "{ \"id\" : " + s.id + ", \"name\" : \"" + s.name + "\" }"))
+            .collect(Collectors.toList());
+
+        return Flowable.fromIterable(stationsAsJson);
+    };
+
+    private static class WeatherStation {
+
+        int id;
+        String name;
+        int averageTemperature;
+
+        public WeatherStation(int id, String name, int averageTemperature) {
+            this.id = id;
+            this.name = name;
+            this.averageTemperature = averageTemperature;
+        }
+    }
+}
+----
+<1> Instruct Reactive Messaging to dispatch the items from the returned `Flowable` to `temperature-values`.
+<2> The method returns a RX Java 2 _stream_ (`Flowable`) emitting a random temperature value every 0.5 seconds.
+<3> Instruct Reactive Messaging to dispatch the items from the returned `Flowable` (static list of weather stations) to `weather-stations`.
+
+The two method each return a _reactive stream_ whose items are sent to the streams named `temperature-values` and `weather-stations`, respectively.
+
+=== Topic Configuration
+
+The two streams are mapped to Kafka Apache topics using the Quarkus configuration file `application.properties`.
+For that, add the following to the file `producer/src/main/resources/application.properties`:
+
+[source]
+----
+kafka.bootstrap.servers=localhost:9092
```

Comment:
Maybe you can add a comment:
# Configure the Kafka broker location

Reply:
Done.